common/src/Utilities/FileUtils.cpp

```diff
@@ -0,0 +1,139 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2021  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "FileUtils.h"
+#include <wx/file.h>
+#include <wx/stdpaths.h>
+#include <wx/filename.h>
+#include "ghc/filesystem.h"
+#include <wx/utils.h>
+
+// ---------------------------------------------------------------------------------
+//  Path namespace
+// ---------------------------------------------------------------------------------
+bool FileUtils::IsRelative(const fs::path &path)
+{
+    return fs::path(path).is_relative();
+}
+
+bool FileUtils::IsDirectoryWithinDirectory(fs::path base, fs::path dir)
+{
+    fs::path relativePath = fs::relative(fs::absolute(dir), fs::absolute(base));
+    if (relativePath.empty())
+    {
+        return false;
+    }
+    return relativePath.string().rfind("..", 0) != 0;
+}
+
+s64 FileUtils::GetFileSize(const fs::path &path)
+{
+    if (!fs::exists(path))
+        return -1;
+    return (s64)fs::file_size(path);
+}
+
+fs::path FileUtils::Normalize(const fs::path &src)
+{
+    if (src.empty())
+    {
+        return fs::path();
+    }
+    return src.lexically_normal();
+}
+
+fs::path FileUtils::MakeAbsolute(const fs::path &src)
+{
+    return ghc::filesystem::absolute(src);
+}
+
+fs::path FileUtils::Combine(const fs::path &srcPath, const fs::path &srcFile)
+{
+    return (srcPath / srcFile).make_preferred();
+}
+
+// Replaces the extension of the file with the one given.
+// This function works for path names as well as file names.
+fs::path FileUtils::ReplaceExtension(const fs::path &src, const fs::path &ext)
+{
+    return fs::path(src).replace_extension(ext);
+}
+
+fs::path FileUtils::ReplaceFilename(const fs::path &src, const fs::path &newfilename)
+{
+    return fs::path (src).replace_filename(newfilename);
+}
+
+fs::path FileUtils::GetFilename(const fs::path &src)
+{
+    return fs::path(src).filename();
+}
+
+fs::path FileUtils::GetFilenameWithoutExt(const fs::path &src)
+{
+    return src.stem();
+}
+
+fs::path FileUtils::GetDirectory(const fs::path &src)
+{
+    return src.parent_path();
+}
```

Comment:
Why are we adding helper functions that just call a single method on the `fs::path`?  Why not just call that method directly?  Future people who know C++ `std::filesystem` but not our special `FileUtils` will probably appreciate if you used the `fs::path` methods

Reply:
https://github.com/PCSX2/pcsx2/blob/6cc969a89ccfa24c6f82a94154855f7934ed7f64/common/src/Utilities/PathUtils.cpp#L125 Going off the original's functions